app/models/check_list.rb

```diff
@@ -300,9 +298,9 @@ def self.refresh_with_observation(observation, options = {})
     end
     if observation && observation.research_grade? && observation.taxon.species_or_lower?
       Rails.logger.info "[INFO #{Time.now}] refresh_with_observation #{observation_id}, adding new listed taxa"
-      if atlas = Atlas.where(taxon_id: observation.taxon_id).first
-        new_place_ids = new_place_ids - atlas.places.map(&:id)
-      end
+      if Atlas.where("taxon_id IN (?)", taxon_ids).count > 0 || CompleteSet.where("taxon_id IN (?) AND place_id IN (?)",taxon_ids, current_place_ids).any?
+        new_place_ids = Place.where(id: new_place_ids).where("admin_level NOT IN (?)",[0,1,2]).pluck(:id) #if under atlas or complete set,
```

Comment:
I nearly missed this comment. We generally put comments on their own lines above the code their commenting on:

```
  # here is a 
  # multiline comment
  do_something_non_obvious()
```